# Задание 1

Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [67]:
import math
import numpy as np

In [68]:
def mean(a):
    return a.sum() / len(a)

In [69]:
# функция расчета коэффициента ковариации со смещенной оценкой ddof=0
def covar(x, y):
    return mean(x * y) - mean(x) * mean(y)

In [70]:
def std(x, ddof=0):
    return math.sqrt(np.sum(np.power(x - mean(x),2)) / (len(x) - ddof))

In [71]:
zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

# считаем коэффициент ковариации со смещенной оценкой ddof=0
COV = covar(zp, ks)
COV_np = np.cov(zp, ks, ddof=0)

print(COV, COV_np)

9157.839999999997 [[ 3494.64  9157.84]
 [ 9157.84 30468.89]]


In [72]:
R1 = COV / (std(zp, 0) * std(ks, 0))
R2 = np.corrcoef(zp, ks)
print(R1, R2)

0.8874900920739158 [[1.         0.88749009]
 [0.88749009 1.        ]]


Коэффициент ковариации = 9157.84
Коэффициент корреляции = 0.88749009

# Задание 2

Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [73]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
a = 0.05
n = len(iq)

# у нас нет данных по стандартному отклонению генеральной совокупности, поэтому используем t-критерий

M = np.mean(iq) # оценка матожидания 

# найдем по таблице t-критерий со степеню свободы 9
t = 2.262

right_boarder = M + (t * np.std(iq) / math.sqrt(n))
left_boarder = M - (t * np.std(iq) / math.sqrt(n))

print('Доверительный интервал от {} до {}'.format(round(left_boarder, 2), round(right_boarder, 2)))

Доверительный интервал от 110.94 до 125.26


# Задание 3

Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для математического
ожидания с надежностью 0.95.

In [74]:
D = 25
n = 27
M = 174.2
a = 0.05

# Так как нам известна дисперсия генеральной совокупности, используем Z-критерий
z = 1.96

right_boarder = M + (z * math.sqrt(D) / math.sqrt(n))
left_boarder = M - (z * math.sqrt(D) / math.sqrt(n))

print('Доверительный интервал от {} до {}'.format(round(left_boarder, 2), round(right_boarder, 2)))

Доверительный интервал от 172.31 до 176.09
